flatbuf: unresolved import `flatbuf_generated::ddlog::__server_api_p2`

I have the following file:
server_api_p2.dl:
```
input relation P2In[string]
output relation P2Out[string]

P2Out[s] :- P2In[s].
```
I create a Rust project as follows:
```bash
ddlog -j -i server_api_p2.dl -L../../lib/
```

And then I compile it with flatbuf support:
```bash
cargo build --features=flatbuf
```

And I get
```bash
$ cargo build --features=flatbuf
>    Compiling server_api_p2 v0.1.0 (differential-datalog/test/datalog_tests/server_api_p2_ddlog)
> error[E0432]: unresolved import `flatbuf_generated::ddlog::__server_api_p2`
>    --> src/flatbuf.rs:575:5
>     |
> 575 | use flatbuf_generated::ddlog::__server_api_p2  as fb;
>     |     ^^^^^^^^^^^^^^^^^^^^^^^^^^---------------^^^^^^^
>     |     |                         |
>     |     |                         help: a similar name exists in the module: `__server_api_p_2`
>     |     no `__server_api_p2` in `flatbuf_generated::ddlog`
> 
> error[E0432]: unresolved import `flatbuf_generated::ddlog::__server_api_p2`
>    --> src/flatbuf.rs:576:35
>     |
> 576 | pub use flatbuf_generated::ddlog::__server_api_p2 ::__String;
>     |                                   ^^^^^^^^^^^^^^^ could not find `__server_api_p2` in `ddlog`
> 
> error: aborting due to 2 previous errors
```

And indeed the module is named `__server_api_p_2` and not `__server_api_p2`.

I think the problem is a mismatch between what `flatc` emits and the code we generate. In particular, it seems that:
```hs
rustFBModule :: (?prog_name :: String) => Doc
rustFBModule = "__" <> (pp $ Casing.quietSnake ?prog_name)
```
which is used for generating the `use` statment has a slightly different interpretation of "snake case" compared to `flatc` when it comes to numbers in the name.

--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -154,6 +154,13 @@ test-imported-souffle-tests7:
     script:
         - (cd test && ./run-souffle-tests-in-batches.py 150 175)
 
+# Test DDlog code from the Declarative Cluster Management project
+test-dcm:
+    extends: .install-ddlog
+    script:
+        - (cd test/datalog_tests &&
+          ./run-test.sh dcm1 debug)
+
 # Torture-test optimized version of redist
 test-redist_opt:
     extends: .install-ddlog

--- a/package.yaml
+++ b/package.yaml
@@ -36,7 +36,6 @@ dependencies:
 - text
 - githash
 - process
-- casing
 - aeson
 
 library:

--- a/src/Language/DifferentialDatalog/FlatBuffer.hs
+++ b/src/Language/DifferentialDatalog/FlatBuffer.hs
@@ -49,7 +49,6 @@ import Control.Monad.Except
 import System.FilePath
 import System.Process
 import GHC.IO.Exception
-import qualified Text.Casing as Casing
 --import Debug.Trace
 import qualified Data.ByteString.Char8 as BS
 
@@ -183,8 +182,39 @@ compileFlatBufferJavaBindings d prog_name =
 jFBPackage :: (?prog_name :: String) => Doc
 jFBPackage = "ddlog.__" <> pp ?prog_name
 
+-- Generate snake-case Rust module name.
+-- replicating the following logic from the `flatc` compiler.
+{-
+std::string MakeSnakeCase(const std::string &in) {
+  std::string s;
+  for (size_t i = 0; i < in.length(); i++) {
+    if (i == 0) {
+      s += static_cast<char>(tolower(in[0]));
+    } else if (in[i] == '_') {
+      s += '_';
+    } else if (!islower(in[i])) {
+      // Prevent duplicate underscores for Upper_Snake_Case strings
+      // and UPPERCASE strings.
+      if (islower(in[i - 1])) {
+        s += '_';
+      }
+      s += static_cast<char>(tolower(in[i]));
+    } else {
+      s += in[i];
+    }
+  }
+  return s;
+}
+-}
 rustFBModule :: (?prog_name :: String) => Doc
-rustFBModule = "__" <> (pp $ Casing.quietSnake ?prog_name)
+rustFBModule = "__" <> char (toLower (head ?prog_name)) <> rustFBModule' (head ?prog_name) (tail ?prog_name)
+
+rustFBModule' :: Char -> String -> Doc
+rustFBModule' _ ""        = ""
+rustFBModule' _ ('_':s)   = char '_' <> rustFBModule' '_' s
+rustFBModule' c (x:s)     | not (isLower x)
+                          = (if isLower c then char '_' else empty) <> char (toLower x) <> rustFBModule' x s
+rustFBModule' _ (x:s)     = char x <> rustFBModule' x s
 
 -- Convert DDlog field name to Java accessor method name,
 -- replicating the following logic from the `flatc` compiler.

--- a/test/datalog_tests/dcm1.dl
+++ b/test/datalog_tests/dcm1.dl
@@ -1,0 +1,77 @@
+import intern
+
+typedef int_t = signed<64>
+
+function sum(g: Group<(int_t)>): (int_t) =
+{
+    var sum: int_t = 0;
+    for (resource in g) {
+        sum = resource + sum
+    };
+    sum
+}
+
+input relation NODE(
+	name: IString, 
+	is_master: bool, 
+	unschedulable: bool, 
+	out_of_disk: bool, 
+	memory_pressure: bool, 
+	disk_pressure: bool,
+	pid_pressure: bool, 
+	ready: bool, 
+	network_unavailable: bool, 
+	cpu_capacity: int_t, 
+	memory_capacity: int_t,
+	ephemeral_storage_capacity: int_t, 
+	pods_capacity: int_t, 
+	cpu_allocatable:int_t, 
+	memory_allocatable: int_t, 
+	ephemeral_storage_allocatable:int_t, 
+	pods_allocatable: int_t) 
+
+input relation POD(
+	pod_name: IString, 
+	status: IString, 
+	controllable__node_name: IString,
+	namespace: IString, 
+	cpu_request: int_t,  
+	memory_request: int_t, 
+	ephemeral_storage_request: int_t,
+	pods_request:int_t,
+	owner_name: IString,
+	creation_timestamp: IString,
+	priority: int_t)
+
+
+output relation CPUUSEDPERNODE(name: IString, cpu: int_t)
+
+output relation MEMORYUSEDPERNODE(name: IString, memory: int_t)
+
+output relation PODSUSEDPERNODE(name:IString, pods:int_t)
+
+output relation SPARECAPACITY(name:IString, cpu_remaining:int_t, memory_remaining: int_t, pods_remaining:int_t) 
+
+
+CPUUSEDPERNODE(node_name, cpu) :-
+	POD(_, _, node_name, _, cpu_request, _, _, _, _, _, _),
+	var cpu = Aggregate((node_name), sum(cpu_request)). 
+	
+MEMORYUSEDPERNODE(node_name, memory) :-
+	POD(pod_name, _, node_name, _, _, memory_request, _, _, _, _, _),
+	var memory = Aggregate((node_name), sum(memory_request)).
+
+PODSUSEDPERNODE(node_name, pods):-
+	POD(pod_name, _, node_name, _, _, _, _, pods_request, _, _, _),
+	var pods = Aggregate((node_name), sum(pods_request)).
+
+SPARECAPACITY(node_name, cpu_remaining, memory_remaining, pods_remaining) :-
+	POD(.controllable__node_name = node_name),
+	(istring_str(node_name) != "null"),
+	NODE(node_name, _, _, _, _, _, _, _, _, _, _, _, _, cpu_allocatable, memory_allocatable, _, pods_allocatable), 
+	CPUUSEDPERNODE(node_name, cpu),
+	MEMORYUSEDPERNODE(node_name, memory),
+	PODSUSEDPERNODE(node_name, pods),
+	var cpu_remaining =  cpu_allocatable-cpu,
+	var memory_remaining =  memory_allocatable-memory,
+	var pods_remaining = pods_allocatable-pods.

